src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/menu/view/MainMenu.kt

```diff
@@ -56,37 +61,35 @@ class MainMenu : MenuBar() {
                 }
                 separator()
                 menu(messages["audioRecorder"]) {
-                    graphic = MaterialIconView(MaterialIcon.MIC, "20px")
+                    graphic = MainMenuStyles.recorderIcon("20px")
                     val pluginToggleGroup = ToggleGroup()
                     viewModel.recorderPlugins.onChange { _ ->
                         items.clear()
-                        items.setAll(viewModel.recorderPlugins.map {
-                            radiomenuitem(it.name) {
-                                userData = it
-                                action { if (isSelected) viewModel.selectRecorder(it) }
+                        items.setAll(viewModel.recorderPlugins.map { pluginData ->
+                            radiomenuitem(pluginData.name) {
+                                userData = pluginData
+                                action { if (isSelected) viewModel.selectRecorder(pluginData) }
                                 toggleGroup = pluginToggleGroup
-                                isSelected = viewModel.selectedRecorderProperty.value == it
-                                viewModel.selectedRecorderProperty.onChange {
-                                    isSelected = viewModel.selectedRecorderProperty.value == it
+                                viewModel.selectedRecorderProperty.toObservable().subscribe {
+                                    isSelected = (it == pluginData)
                                 }
                             }
                         })
                     }
 
                 }
                 menu(messages["audioEditor"]) {
-                    graphic = MaterialIconView(MaterialIcon.MODE_EDIT, "20px")
+                    graphic = MainMenuStyles.editorIcon("20px")
                     val pluginToggleGroup = ToggleGroup()
                     viewModel.editorPlugins.onChange { _ ->
                         items.clear()
-                        items.setAll(viewModel.editorPlugins.map {
-                            radiomenuitem(it.name) {
-                                userData = it
-                                action { if (isSelected) viewModel.selectEditor(it) }
+                        items.setAll(viewModel.editorPlugins.map { pluginData ->
+                            radiomenuitem(pluginData.name) {
+                                userData = pluginData
+                                action { if (isSelected) viewModel.selectEditor(pluginData) }
                                 toggleGroup = pluginToggleGroup
```

Comment:
are these toggleGroup ever set to something other than an empty `ToggleGroup()`

Reply:
i see `pluginToggleGroup = ToggleGroup` but it is never changed after that? Am i missing something

Reply:
I think what's going on there is I'm creating a new toggle group, then assigning the radio items to that group. That way the radio items know only one in the group should be selected.

Reply:
Oh ok i see now. You did this in lue of `radiomenuitem(pluginData.name, pluginToggleGroup)`

Reply:
Yes, but really what you have there is better, I think

Reply:
It does make whats going on easier to see